### PR TITLE
PODAUTO-199: Upstream rebase to VPA 1.1.2

### PR DESCRIPTION
**Pull request by maxcao13**
These steps were followed in these commits:
```
git merge-base kubernetes/master vertical-pod-autoscaler-1.1.2
git log --oneline 151923f..vertical-pod-autoscaler-1.1.2  | grep -v Merge.pull
```
The first one is to find the common ancestor, and the next one to show what has changed from that point to the 1.1.2 release (ignoring merge commits)

Then we cherry-pick VPA related commits:
```
for commit in 7d2bd3037 0bc73b3fb 991308df3 cd673e325 b4ce36e40 3609234c5; do
  git cherry-pick $commit || break
  git commit --amend -m "UPSTREAM: <drop>: $(git show --format=format:%s -s $commit)"
done
```

### CODE REVIEW

**Comment by maxcao13** [comment]
@frobware @enxebre Hi, can someone from the cloud team review this and see if it's okay from your pov? if not, @joelsmith mentioned as an alternative would be for us to 

> identify all the VPA-specific commits from the merge point until the VPA release and cherry-pick them as UPSTREAM: drop: commits (i.e., they would only be in our repo until the next rebase which would bring the actual original upstream commits in.

Thanks!

**Comment by jkyros** [comment]
So I'm definitely not the expert here, and I don't want to muddy the water, but seems like we might not want to rebase to master: 
- This purportedly moves us to upstream vpa 1.1.2, but it looks like the vpa version is still 1.1.0: https://github.com/openshift/kubernetes-autoscaler/blob/d2da86ef13f82632a2cf4c83b3b7f2bc8d1ae0b9/vertical-pod-autoscaler/common/version.go#L20
- https://github.com/openshift/kubernetes-autoscaler/pull/306 merged `upstream/cluster-autoscaler-1.30.1`, whereas this merges `master`, and `master` (among other things) seems to drag the cluster-autoscaler deps up to 1.31 which we might not want yet: https://github.com/openshift/kubernetes-autoscaler/blob/d2da86ef13f82632a2cf4c83b3b7f2bc8d1ae0b9/cluster-autoscaler/go.mod#L221
- There is a tag for VPA 1.1.2 upstream https://github.com/kubernetes/autoscaler/tree/vertical-pod-autoscaler-1.1.2, which would seem to be what we want, but it hasn't merged to master yet. I'm assuming this is for lifecycle/kubernetes release reasons and they didn't screw up and it's not a mistake :smile: 
- The "release code that isn't in master yet" seems normal (or, well, consistent at least for this cycle) because the cluster autoscaler tag we rebased to in #306 also doesn't look like it's reflected in master yet? (or I'm missing something, please tell me if I am a quack) 

Diffs for convenience: 

- OpenShift vs upstream vertical-pod-autoscaler-1.1.2: https://github.com/openshift/kubernetes-autoscaler/compare/master...kubernetes:autoscaler:vertical-pod-autoscaler-1.1.2
- Upstream `master` vs upstream `vertical-pod-autoscaler-1.1.2`: https://github.com/kubernetes/autoscaler/compare/master...vertical-pod-autoscaler-1.1.2
- Upstream `master` vs `cluster-autoscaler-1.30.1`: https://github.com/kubernetes/autoscaler/compare/master...kubernetes:autoscaler:cluster-autoscaler-1.30.1
- Upstream `cluster-autoscaler-1.30.1`  vs  `vertical-pod-autoscaler-1.1.2`: https://github.com/kubernetes/autoscaler/compare/cluster-autoscaler-1.30.1...vertical-pod-autoscaler-1.1.2

The right thing to would _seem_ to be rebase to the `vertical-pod-autoscaler-1.1.2` tag but I'll defer to someone more expert-y :smile: 

TLDR: upstream release tags seem to be tracking different stuff than master, suspect that might be what we want if we want VPA 1.1.2?

**Comment by maxcao13** [comment]
We've decided to cherry pick VPA related commits from upstream/master since the last rebase instead of interfering with clusterautoscaler related things, so this pr is a lot smaller. We also tag them as drop commits so that they can be dropped in the next rebase.

**Comment by joelsmith** [comment]
/lgtm
/approve

The failing tests are specific to cluster autoscaler, so I think we need to find a staff engineer to override them since they aren't applicable to these VPA changes.
